src/navigation/checkout/Restaurant.js

```diff
@@ -46,121 +64,226 @@ const LoadingPhantom = (props) =>
+    });
+  }
 
-  return (
-    <SafeAreaView style={{ flex: 1 }} edges={['bottom']}>
-      <View style={{ flex: 1, paddingTop: 60 }}>
-        <View style={{ position: 'absolute', top: 0, left: 0, width: '100%', height: 60 }}>
-          <GroupImageHeader image={ restaurant.image } text={restaurant.name} onInfo={() => setInfoModal(true)} category={ restaurant.facets.category } />
-        </View>
-        {renderWarningBanner()}
-        { isLoading && <Center w="100%">
-          <LoadingPhantom color={'cyan.200'} />
-          <LoadingPhantom color={'gray.200'} />
-          <LoadingPhantom color={'amber.200'} />
-        </Center> }
-        { !isLoading && <Menu
-          restaurant={ restaurant }
-          menu={ data }
-          onItemClick={ menuItem => navigate('CheckoutProductDetails', { product: menuItem, restaurant }) }
-          isItemLoading={ menuItem => props.loadingItems.includes(menuItem.identifier) } /> }
+  const renderRestaurantProfile = () => (
+    <RestaurantProfile
+      onInfo={() => setInfoModal(true)}
+      restaurant={restaurant}
+    />
+  );
+
+  const renderRestaurantMenuHeader = () => (
+    <RestaurantMenuHeader
+      activeSection={activeSection}
+      sectionRef={sectionListRef}
+      sections={sections}
+      isLoading={isLoading}
+    />
+  );
+
+  const renderSection = section => (
+    <>
+      <View id={'section-' + section.index}>
+        <Text
+          style={{
+            fontSize: 18,
+            // fontWeight: 'bold',
+            marginHorizontal: 16,
+            marginTop: 48,
+            marginBottom: 8,
+          }}>
+          {section.title}
+        </Text>
+        {section.data.map((item, innerIndex) => (
+          <RestaurantMenuItem
+            item={item}
+            isLoading={isLoading}
+            key={innerIndex}
+            onPress={menuItem =>
+              navigate('CheckoutProductDetails', {
+                product: menuItem,
+                restaurant,
+              })
+            }
+          />
+        ))}
       </View>
-      { showFooter ? (
+    </>
+  );
+
+  const renderEmptyFooter = () => (
+    <View style={{height: screenHeight * 0.4}}></View>
+  );
+
+  const renderFunctions = [
+    renderRestaurantProfile,
+    renderWarningBanner,
+    renderRestaurantMenuHeader,
+    ...sections.map(section => () => renderSection(section)),
+    renderEmptyFooter,
+  ];
+
+  const listRenderItem = ({item, index}) => {
+    return renderFunctions[index] ? renderFunctions[index]() : null;
+  };
+
+  return (
+    <View
+      style={{
+        display: 'flex',
+        width: '100%',
+      }}>
+      <FlatList
+        stickyHeaderIndices={[2]}
+        data={Array.from({length: renderFunctions.length}, (_, index) => index)}
+        renderItem={listRenderItem}
+        ref={sectionListRef}
+        viewabilityConfig={viewabilityConfig.current}
+        onViewableItemsChanged={handleViewableItemsChanged.current}
+      />
+      {showFooter ? (
```

Comment:
Non sure what happened with a footer / Cart button, but I don't see it:

![Simulator Screenshot - iPhone SE (3rd generation) - 2024-03-14 at 15 24 48](https://github.com/coopcycle/coopcycle-app/assets/70273239/f8de05b4-aaa7-4bc1-bfb8-272132d38ee7)

![Simulator Screenshot - iPhone SE (3rd generation) - 2024-03-14 at 15 23 52](https://github.com/coopcycle/coopcycle-app/assets/70273239/d858f4af-f6e7-43a1-b7e0-a203a0af2545)